File: Data Structures/Linked Lists/Queues/1700/1700.py
# Define a class for the solution
class Solution:
    # Define a method to count the number of students unable to eat lunch
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        # Initialize the result as the total number of students
        res = len(students)
        # Count the number of students who prefer each type of sandwich
        cnt = Counter(students)

        # Iterate over the sandwiches in the stack
        for sandwich in sandwiches:
            # If there are students who prefer the current sandwich
            if cnt[sandwich] > 0:
                # Decrease the number of students unable to eat lunch
                res -= 1
                # Decrease the number of students who prefer the current sandwich
                cnt[sandwich] -= 1
            else:
                # If no students prefer the current sandwich, return the result
                return res

        # Return the result
        return res

# Time complexity: O(n), where n is the number of students
# Space complexity: O(n), where n is the number of students

File: Data Structures/Linked Lists/Queues/1700/1700.go
package main

func countStudents(students []int, sandwiches []int) int {
	var count int
	for len(students) > 0 {
		if students[0] == sandwiches[0] {
			// The student can eat the sandwich
			students = students[1:]     // Remove the student from the queue
			sandwiches = sandwiches[1:] // Remove the sandwich from the stack
			count = 0                   // Reset count since a student ate a sandwich
		} else {
			// The student goes to the end of the queue
			students = append(students[1:], students[0])
			count++ // Increment count since no student ate a sandwich
		}
		// If all students have been moved to the end of the queue without eating
		if count == len(students) {
			break
		}
	}
	return len(students) // The number of students unable to eat
}
